### DISCUSSION TITLE
A good mechanism to share & find casaos "custom install"

### DISCUSSION THREAD

**Discussion by MX10-AC2N** [IceWhaleTech/CasaOS#439] [General]
Hello to all,
I'm opening a new discussion about sharing information about custom installs in casaos.
Do you think it would be wise to open a kind of wiki or forum where we could detail the installation of some Docker, while adding photos, descriptions of apps and different possible settings in the docker "settings" of casaos.
This would be much easier to find and share, currently I browse the Discord discussions to find answers to the problems I encounter and it is not really simple. 
There is https://appfile.casaos.io/community but there is no description, nor details on the possible settings just a file to download... And no possibility to create a new custom install
What do you think about it ?
Thanks again to all the 'Devs' team, you do a great job! And thanks to all for sharing...
Have a nice day.

**Comment by JohnGuan** [comment]
Thanks! I think it's a very good idea and topic!
@tigerinus , maybe this will be the start of our wiki?

I think a tutorial + a place to share the appfile & detailed instructions would be a quick way to start this thing.

**Reply by tigerinus** [reply]
I believe this is a perfect usage of Wiki for CasaOS. We could have a page to allow people share experiences about custom installs.

https://wiki.casaos.io is just setup - maybe it's time for poeple to flood in and get relevant content populated. 😁

**Reply by MX10-AC2N** [reply]
Hi

Thank you @JohnGuan @tigerinus for your answers.

I have looked https://wiki.casaos.io/ it seems to be able to do the job but for now there is no possibility to create an account on the site just a login page. It would also be necessary to add a "custom install" tab.

But I wonder if this is really the easiest way for CasaOs users (more or less beginners) to share their custom installs, the Wiki requires a particular syntax, not everyone practices this syntax and so I wonder if a Forum would not be more suitable for use by novice users. Moreover it would allow a more varied tree of subjects, it would be also perhaps more beautiful visually than a Wiki. 

Thanks again.

Translated with www.DeepL.com/Translator (free version)

**Reply by JohnGuan** [reply]
@MX10-AC2N 
For an introductory tutorial on GitHub, on Wiki.js or anywhere seems to work.
For sharing appfiles, do you think GitHub Discussions is an easy option to get started?

**Reply by tigerinus** [reply]
I've enabled GitHub login for wiki.casaos.io.

The syntax is in Markdown language. If you know how to create a README.md you know how to edit a Wiki page. 😀

**Reply by Hutch79** [reply]
> But I wonder if this is really the easiest way for CasaOs users (more or less beginners) to share their custom installs, the Wiki requires a particular syntax, not everyone practices this syntax and so I wonder if a Forum would not be more suitable for use by novice users.

I could really recommend [discourse](https://www.discourse.org/).
It is very simple to use, widely spread and has grate features like a variety of login services like GitHub, Google and much more.

It also supports Markdown, HTML and bbcode, or you just use the menu to tweak your text to your liking. So it's very user-friendly.

**Reply by MX10-AC2N** [reply]
Thank @Hutch79 
Very good proposal.
Discourse seems to be a good choice as it will be easier than a Wiki and there could be various topics around the different IceWhale Tech projects. (zimaboard, casaos, cyberise, ...)

Thanks again

**Reply by Hutch79** [reply]
There is also a wiki/docs plugin for Discourse. See here: https://meta.discourse.org/docs
Here is also a list of all Plugins available for it: https://www.discourse.org/plugins

**Reply by JohnGuan** [reply]
@Hutch79 Thank you for the recommendation, it was great.
I've been looking at Discourse as well, but I made a huge mistake in a previous project. Which is I thought there was a need for a lot of people to discuss it, then I opened a Discourse and I ended up being the only one using it for over a year.

I think I'll see if the community really needs to discuss it through more channels. The problem at the moment is that we now have limited activity even with just Discord and GitHub. This of course has a lot to do with my operations work.
If GitHub's Discussions are up to the task of communication, I would want to start such topics here first. When there are more needs than can be met, I'll migrate to any of the appropriate platforms.

What do you think? @MX10-AC2N @tigerinus @Hutch79 and others

**Reply by Hutch79** [reply]
Even if the Community does not use Discourse as a Community platform, I guess it would be an easy way for everyone to create Wiki articles.

**Reply by JohnGuan** [reply]
> Even if the Community does not use Discourse as a Community platform, I guess it would be an easy way for everyone to create Wiki articles.

I see your point. Easy participation is important.
This is also a thing I've been paying careful attention to in the IceWhale/CasaOS community.

**Reply by MX10-AC2N** [reply]
I agree with @Hutch79 
and I think that there will be a few people to keep the discourse alive, considering the number that we are on discord.
That said the fact of being able to open the discussions on Github it is Top but it should rather be reserved for the requests and ideas related to the developments, a contrario on the discourse we would have only post on the use and the various possibilities of parameter setting made by the users, reserved that for the operation of CasaOs and the Zimaboard and the future projects...
In any case it is a pleasure to share with you..

**Reply by Hutch79** [reply]
Just had something in my mind.
If Discourse for some reason isn't a solution, [Flarum](https://flarum.org/) looks also grate. Although I would recommend Discourse because they have way more to tweak around with permissions and stuff.

One benefit does Flarum have, it runs on nearly every webspace, where Discourse needs Docker.

**Reply by JohnGuan** [reply]
> If Discourse for some reason isn't a solution, [Flarum](https://flarum.org/) looks also grate. Although I would recommend Discourse because they have way more to tweak around with permissions and stuff.

Almost forgot about this, it's a great choice too.

**Comment by MX10-AC2N** [comment]
@tigerinus 
It's alright, I can login in Wiki..
Is there a way for the Wiki to be multilingual, for now there is only English, that is a bit limited.
I'm not an expert in markdown at all but I have to learn it.
Thanks again.

**Reply by tigerinus** [reply]
We are still ramping up in terms of managing and operating this Wiki, so it's going to be English only for now. Once things go more smoothly, we will consider adding multi language support. What language do you use?

**Reply by MX10-AC2N** [reply]
I speak french..

**Comment by JohnGuan** [comment]
@MX10-AC2N @Hutch79 
Would you guys mind writing a draft about such content?
Let's base on these drafts and discuss where they should be placed and how to guide other community members to access and contribute.

**Reply by Hutch79** [reply]
I could try.
Basically a draft on which information you should provide about your appfile, right?
So, like a GitHub Bug report template but for appfiles.

**Reply by Hutch79** [reply]
# Appfile Title (*Nginx Proxy Manager*)

### Short description about the Software you are providing
*Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.*

### Information
- Docker Image ID: *jc21/nginx-proxy-manager:latest*
- Version: *latest*
- Dependencies: *MySQL*
- Ports: *81, Web interface - 80, HTTP Proxy - 443, HTTPS Proxy*
- 
All the stuff you set up and/or changed

### Support/Creator
Link to Docker Maintainer: (Link to creator website)
Support Website: (Wiki/docs, support forum, just everything you can find)

---
At the moment, it's all that comes to my mind.
All *Italic* texts are placeholders

**Reply by JohnGuan** [reply]
I also wrote a draft with reference to what you wrote.
https://github.com/IceWhaleTech/ZimaDocs/discussions/3

**Reply by Hutch79** [reply]
That looks grate!
Also, it has some stuff I forgot about ^^'

Basically, it is a nicer presentation of the [appfile ](https://github.com/IceWhaleTech/CasaOS-AppStore/blob/main/Apps/Jellyfin/appfile.json)content.

**Comment by JohnGuan** [comment]
@tigerinus @MX10-AC2N @Hutch79 @Hedius @xinity @dtaivpp
I've written a draft in the below link. if you have any suggestions or ideas. Feel free to comment directly. I will modify and optimize the draft based on everyone's comments.
If there are any deviations, please point them out promptly.
- https://github.com/IceWhaleTech/ZimaDocs/discussions/3

**Comment by MX10-AC2N** [comment]
Hi friends,

This is Top what was proposed...

 I also tried to make one for PhotoPrism with MariaDB but I'm not really satisfied, the formatting is not obvious, it took me a lot of time and I'm not an expert in Markdown and the result does not seem very clear.

=> https://github.com/MX10-AC2N/CasaOs-Appfile/blob/main/photoprism_mariadb.md#photoprism-with-mariadb-custom-install

**Reply by JohnGuan** [reply]
We are basically the same.
Did you read the draft I wrote? Is there enough information?
- https://github.com/IceWhaleTech/ZimaDocs/discussions/3

**Comment by MX10-AC2N** [comment]
Hi 
You are right our proposals are quite similar and what you did with Jellyfin is really good, there is I think everything you need to install and explain the different configuration possibilities. 
In my example, where I needed to install 2 dockers in parallel Photoprism and MariaDB it would be interesting to be able to link these 2 dockers so that the Photoprism docker can't be started until the MariaDB_Photoprism docker is running but that's another subject, there are some things to work on because having the databases in docker visible in the CasaOs web interface takes space for nothing. While I don't have a tile to access cups or scanservjs, I have 2 tiles mariadb_nextcloud and mariadb_photoprism that are only useful for their respective application, these tiles could be underlayed with the docker tiles of the applications they are linked to
Wouldn't it be possible to integrate the installation of Mariadb directly into the docker of its different applications. Sorry, I'm still a beginner with Docker and maybe I'm not doing things correctly.
Otherwise I think that a discourse would be interesting to set up because it will be more accessible to all to exchange than Github...
Thanks again
![Capture d’écran (70)](https://user-images.githubusercontent.com/62643837/186615431-f7eb7b90-23d1-47e5-8946-8213f3cc79dc.png)
![Capture d’écran (70)_LI](https://user-images.githubusercontent.com/62643837/186616954-17fb08c5-a534-4940-997d-e8c7f92dda0c.jpg)

**Reply by JohnGuan** [reply]
I get it. Different installation methods, such as certain containers that require a specified database. Or apps that collaborate between multiple containers are also required to be shown in such shares.
I've been testing some of these systems recently, and it's worth exploring.
Thanks for your advice and try!

**Reply by Hutch79** [reply]
Maybe implement a folder system for the desktop?

**Comment by JohnGuan** [comment]
@MX10-AC2N @Hutch79 
We are developing a proper documentation system.
You can share with me what you want to publish in any form you like. Office, Google Docs, or whatever. We will then assist with editing and publishing and attribution.
If everything works well, we will further accelerate the development of a way to facilitate submissions.
My email: john@icewhale.org
